### PR TITLE
Deprecate `Localization.getLocalizedStringWithNamespace`

### PR DESCRIPTION
**Pull request by mattbjordan**
`Localization.getLocalizedStringWithNamespace` will be [removed in 4.0](https://github.com/iTwin/itwinjs-backlog/issues/513), but for now it [should be deprecated](https://github.com/iTwin/itwinjs-core/pull/4630#discussion_r1020048294).

This is my first time deprecating anything, so let me know if there's any step that I missed.

### CODE REVIEW

**Comment by pmconne** [comment]
Add to "Deprecations" section of NextVersion.md.

File: core/frontend/src/tools/Tool.ts [hunk 1]
```diff
@@ -394,6 +394,7 @@ export class Tool {
 
   private static getLocalizedKey(name: string): string | undefined {
     const key = `tools.${this.toolId}.${name}`;
+    // eslint-disable-next-line deprecation/deprecation
     const val = IModelApp.localization.getLocalizedStringWithNamespace(this.namespace, key);
```
Comment (pmconne) [inline core/frontend/src/tools/Tool.ts:398]: Replacing this call with one of the suggestions in your `@deprecated` message isn't an option?

Comment (mattbjordan) [reply core/frontend/src/tools/Tool.ts:398]: I can absolutely do that for each place that I added the eslint-disabled line and get the changes over with now. I was unsure whether to leave this for later or do it now.

Comment (pmconne) [reply core/frontend/src/tools/Tool.ts:398]: Yeah, generally if we tell people to stop using an API we should stop using it too.

**Review by pmconne — APPROVED** [review]

**Review by raplemie — APPROVED** [review]
LGTM

**Review by calebmshafer — APPROVED** [review]
LGTM

**Pull request merged**